03week/towersOfHanoi.js

```diff
@@ -19,24 +19,46 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+function movePiece(startStack, endStack) {
+  stacks[endStack].push(stacks[startStack].pop());
 }
 
-function isLegal() {
-  // Your code here
-
+function isLegal(startStack, endStack) {
+  if(startStack.length === 0){
+    console.log('This is an empty array, choose another');
+    return false;
+  } else if(endStack.length === 0){
+    return true;
+  } else if(stacks[startStack][stacks[startStack].length - 1] > stacks[endStack][stacks[endStack].length - 1]){
+    console.log('This is illegal!  Please choose another stack');
+    return false;
+  } else {
+    return true;
+  }
 }
 
 function checkForWin() {
-  // Your code here
+  if(stacks.b.length === 4 || stacks.c.length === 4){
+    console.log('Congratulations!  You win!');
+  } else {
+    console.log('Make another move.');
+  }
 
 }
 
 function towersOfHanoi(startStack, endStack) {
-  // Your code here
-
+  startStack = startStack.toLowerCase().trim();
+  endStack = endStack.toLowerCase().trim();
+  if((startStack === 'a' || startStack === 'b' || startStack === 'c') && (endStack === 'a' || endStack === 'b' || endStack === 'c')){
```

Comment:
this evaluation should be extracted into another function that would check if it's one of these letters. Also think of a way to check for a letter without strict checking if the variables are a, b, or c